sns/parity_math.c

```diff
@@ -32,7 +32,18 @@
 #define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_SNS
 #include "lib/trace.h"
 
-#define ir_invalid_col_t UINT8_MAX
+#define ir_invalid_col_t   UINT8_MAX
+
+#ifndef __KERNEL__
+#define ALLOC_ARR_INFO(arr, nr, msg, ret) ({					\
+	M0_ALLOC_ARR(arr, nr);							\
+	if (arr == NULL) 							\
+		(ret) = M0_ERR_INFO(-ENOMEM,					\
+				    "failed to allocate memory for " msg);	\
+	else 									\
+		(ret) = M0_RC_INFO(0, "allocate memory for " msg);		\
+})
```

Comment:
I know this will simplify some of the code.
But to me, it is not necessary.
Error message (generic alloc failure) is already log in m0_alloc().